src/Engine/GameAction.java

```diff
@@ -268,7 +276,12 @@ public GameEventQueue getEvents(MapMaster gameMap)
       // If our guy can't survive there, end him.
       if( !unit.model.propulsion.canTraverse(gameMap.getEnvironment(unitDestination)) )
       {
+        ArrayList<Unit> ary = new ArrayList<Unit>();
+        ary.add(unit);
+        boolean fatal = false;
```

Comment:
Needs to be true, or you will get 1 less HP's worth of charge than you should.

Reply:
Gah. I had it that way, and changed it for testing, and forgot to change it back. Good catch.